components/ws-manager-api/core.proto

```diff
@@ -124,6 +127,29 @@ enum StopWorkspacePolicy {
     ABORT = 2;
 }
 
+enum StopWorkspaceReason {
+    // The user stopped the workspace
+    USER_STOPPED = 0;
+
+    // The user deleted the workspace
+    USER_DELETED = 1;
+
+    // The user cancelled the running prebuild
+    USER_CANCELLED_PREBUILD = 2;
+
+    // An admin force-stopped the workspace
+    ADMIN_STOPPED = 3;
+
+    // An admin blocked the user
+    ADMIN_BLOCKED = 4;
```

Comment:
IMO, we shouldn't encode if an Admin did it or not. This assumes that there's a very binary distinction between user and admin (which isnt' the case, and won't be in the future either). Instead, if we want the identity we should also specify it as a field in the request - `StopRequestedBy = milan`

Reply:
Ultimately this is up to Workspace team, but my reasoning is based on this:

* We want to use the enum to categorise classes of reasons, not every single reason. This aids composability but also reduces decision paralysis for which category to use.
* We _should_ have an extra field for the reason which explains it for the human. This can also include information such as if an admin did it, or if it's due to lack of credits. The system wouldn't use this string directly to make decisions, it would only store it to show the user.
* The enums here should not account for every possible way that server can use the StopWorkspace RPC. Instead, it needs to consider the API domain from the `ws-manager` perspective. We do not want to leak concepts such as credits or billing into here, nor do we want to leak concepts like admin or regular user - those belong into the WebApp domain model but not necessarily into ws-manager.
* We can always add reasons here, so if there's a sufficient reason to expand, we always can. Removing is a lot more difficult.